catalog/plugin/catalog-plugin-metacardbackup-filestorage/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/storage/filestorage/MetacardBackupFileStorage.java

```diff
@@ -0,0 +1,210 @@
+    public String getOrganization() {
+        return (String) describableProperties.get(ORGANIZATION);
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public void deleteData(String id) throws IOException, MetacardBackupException {
+        if (StringUtils.isEmpty(outputDirectory)) {
+            throw new MetacardBackupException(
+                    "Unable to delete stored data; no output directory specified.");
+        }
+
+        deleteBackupIfPresent(id);
+    }
+
+    public void storeData(String id, byte[] data) throws IOException, MetacardBackupException {
+        if (StringUtils.isEmpty(outputDirectory)) {
+            throw new MetacardBackupException("Unable to store data; no output directory specified.");
+        }
+
+        if (data == null) {
+            throw new MetacardBackupException("No data to store");
+        }
+
+        Path metacardPath = getMetacardDirectory(id);
+        if (metacardPath == null) {
+            String message = String.format("Unable to create metacard path directory for %s", id);
+            LOGGER.debug(message);
+            throw new MetacardBackupException(message);
+        }
+
+        try {
+            Path parent = metacardPath.getParent();
+            if (parent != null) {
+                Files.createDirectories(parent);
+            }
+            Files.createFile(metacardPath);
+        } catch (IOException e) {
+            LOGGER.debug("Unable to create backup file {}.  File may already exist.",
+                    metacardPath,
+                    e);
+        }
+
+        try (OutputStream outputStream = new FileOutputStream(metacardPath.toFile())) {
+            IOUtils.write(data, outputStream);
+        }
+    }
+
+    public void setOutputDirectory(String outputDirectory) {
+        this.outputDirectory = outputDirectory;
+    }
+
+    public String getOutputDirectory() {
+        return outputDirectory;
+    }
+
+    public void refresh(Map<String, Object> properties) {
+        Object outputDirectory = properties.get(OUTPUT_DIRECTORY_PROPERTY);
+        if (outputDirectory instanceof String && StringUtils.isNotBlank((String) outputDirectory)) {
+            this.outputDirectory = (String) outputDirectory;
+            LOGGER.debug("Updating {} with {}", OUTPUT_DIRECTORY_PROPERTY, outputDirectory);
+        }
+
+        Object id = properties.get(ID);
+        if (id instanceof String) {
+            setId((String) id);
+        }
+    }
+
+    private void deleteBackupIfPresent(String filename) throws MetacardBackupException {
+        Path metacardPath = getMetacardDirectory(filename);
+        if (metacardPath == null) {
+            LOGGER.trace("Unable to delete backup for: {}", filename);
+            throw new MetacardBackupException("Unable to delete backup");
+        }
+
+        try {
+            Files.deleteIfExists(metacardPath);
+            while (metacardPath.getParent() != null && !metacardPath.getParent()
```

Comment:
This makes me a little nervous. What if metacardPath and outputDirectory where the same logical path, but differed because of a symbolic link? Would this could keep climbing the directory tree and delete everything? Can that ever happen?